screens/HomeScreen.js

```diff
@@ -8,25 +8,28 @@ import * as WebBrowser from 'expo-web-browser';
 import { MonoText } from '../components/StyledText';
 import { TextList } from '../components';
 
-const ALL_EXPENSES = [
-  { id: 1, name: 'Buy a book', amount: 20 },
-  { id: 2, name: 'Buy dog food', amount: 5 },
-  { id: 3, name: 'Book a flight ticket', amount: 225 },
-  { id: 4, name: 'Buy speakers', amount: 300 },
-  { id: 5, name: 'Book a concert ticket', amount: 50 },
-  { id: 6, name: 'Buy birthday present', amount: 100 },
-  { id: 7, name: 'Buy boots', amount: 400 },
-];
-
 export default function HomeScreen() {
-  const [expenses, setExpenses] = useState(ALL_EXPENSES);
+  const [expenses, setExpenses] = useState([]);
+
+  const initializeData = async () => {
+    const storedExpenses = JSON.parse(await AsyncStorage.getItem('expenses'));
+    setExpenses(storedExpenses ? storedExpenses : []);
+    console.log('stored Expenses', storedExpenses);
+  };
+  useEffect(() => {
+    initializeData();
+  }, []);
+
   const [name, setName] = useState('');
   const [amount, setAmount] = useState('');
 
   const handleSubmitForm = () => {
     if (name !== '' && amount > 0) {
       const expense = { name, amount };
       setExpenses([...expenses, expense]);
+
+      AsyncStorage.setItem('expenses', JSON.stringify(expenses));
```

Comment:
> TODO : The app isn't behaving the way it should , when refreshing all data should still be visible on the list but last item is not

Think about thee questions in this method. 
1. What is shown on the screen?
2. What is stored to AsyncStorage?
3. What will be visible on next app start?

Reply:
ok best. right.

next please log what you save in the components state. this should be all items we see on the screen.
then log what you store to AsyncStorage. this should be all items except the last item, right?

Reply:
yess, same as last week , last item visible on the list but when i log it's not visible

Reply:
what have you logged? please show it. otherwise I cannot understand fully to talk about the right.

Reply:
<img width="980" alt="Screenshot 2020-04-15 at 14 37 10" src="https://user-images.githubusercontent.com/43641321/79337883-a01d9700-7f26-11ea-9572-adc9006097ef.png">

Reply:
in the first case for state, you are not logging what you are setting.
what happens, if you put `expenses` into `setExpenses`?

Reply:
it's immediately visible on my list ..

Reply:
> it's immediately visible on my list ..

Is it?
This cannot happen for `setExpenses(expenses)`.
I meant that. When you log `console.log('state', expenses)` than see what happens if you set `expenses`.

> but my list is showing whats is in state , meaning whats in expenses
Correct. In the render part `expenses` is fully updated before re-rendering.

Reply:
the data we want to store is `expenses` with the missing new `expense`.

```
console.log([…expenses, expense]);
setExpenses([…expenses, expense]);
console.log([…expenses, expense]);
AsyncStorage.setItem('expenses', JSON.stringify([…expenses, expense]));
```

imagine `[…expenses, expense]`  as your updated expenses. this is what you want everywhere. 
best would be to have a `const` for it, so it will be more clear:

```
const updatedExpenses = […expenses, expense];

console.log(updatedExpenses);
setExpenses(updatedExpenses);
console.log(updatedExpenses); // this second log is redundant an not necessary, because it does exactly the same as two lines before
AsyncStorage.setItem('expenses', JSON.stringify(updatedExpenses));
```

setting exactly the same data to state and storage with that code.